src/ui/src/app/deployments/deployments.component.ts

```diff
@@ -22,26 +27,80 @@ export class DeploymentsComponent implements OnInit {
     private config: ConfigService,
     private mdIconRegistry: MdIconRegistry,
     private sanitizer: DomSanitizer
-  ){
-    mdIconRegistry
-      .addSvgIcon('layers',
-        sanitizer.bypassSecurityTrustResourceUrl(`/assets/icons/layers.svg`));
-  }
+  ) {}
 
   ngOnInit() {
+    this.mdIconRegistry.addSvgIcon(
+      'search',
+      this.sanitizer.bypassSecurityTrustResourceUrl(`/assets/icons/search.svg`)
+    );
     // Do not show the page if the feature is not enabled
-    if(!this.config.releasesEnabled) {
+    if (!this.config.releasesEnabled) {
       return this.router.navigate(['/404']);
     }
     this.loadDeployments();
   }
 
   loadDeployments(): void {
-    this.deploymentsService.getDeployments()
-    .finally(()=> {
-      this.loading = false;
-    }).subscribe(deployments => {
-      this.deployments = deployments;
-    })
+    this.deploymentsService
+      .getDeployments()
+      .finally(() => {
+        this.loading = false;
+      })
+      .subscribe(deployments => {
+        this.deployments = deployments;
+        this.filterDeployments();
+        this.namespaces = this.exportNamespaces();
+      });
+  }
+
+  exportNamespaces(): string[] {
+    var list: string[] = ['All'];
+    this.deployments.forEach(dp => {
+      if (list.indexOf(dp.attributes.namespace) == -1) {
+        list.push(dp.attributes.namespace);
+      }
+    });
+    return list;
+  }
+
+  filterDeployments() {
```

Comment:
I think we can create a common filter component that generate the `select` elements and trigger an event when some of them change.